platform/engineering/design-docs/2020-12-10-auto-deploy-cms-build.md

```diff
@@ -0,0 +1,280 @@
+
+The VA.gov website can be deployed to production through the partial (or content-only) deploy or the full deploy.
+
+The **full deploy** builds from _the latest commit of `vets-website`_ and the latest content downloaded from the Drupal server.
+
+The **partial, content-only deploy** builds from the _latest release-tagged commit of `vets-website`_ and the latest content downloaded from the Drupal server.
+
+Both deploys share the same Jenkins pipeline, described by the `deploys/warn-release-deploy` [Jenkinsfile](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/Jenkinsfiles/deploys/warn-release-deploy) in the `devops` repo.
+- There is a `use_latest_release` flag passed as a parameter to the pipeline that determines which deploy is run.
+  - This value is true for partial deploys in order to build from the latest release.
+  - This value is false for full deploys in order to build from the latest commit.
+- If the commit used in either deploy failed its build, the deploy will also fail.
+- Either deploy can be manually invoked, independently of the daily schedule.
+
+There is an existing [automatic full deploy job scheduled to run at 2:00pm ET daily](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/jenkins-vetsgov/seed_job.groovy#L255-L294). It runs an additional [pre-release job](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/jenkins-vetsgov/seed_job.groovy#L250) ([`builds/vets-website-content-vagovprod`](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/jenkins-vetsgov/seed_job.groovy)) defined in [`Jenkinsfile.content`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/Jenkinsfile.content) in the `vets-website` repo.
+
+To deploy content changes at will, that "auto-deploy" job is manually triggered with the `use_latest_release` flag by the CMS team or CMS users with the `Content Admin` or `Content Publisher` roles.
+
+#### Pain points
+
+- Content writers want to quickly draft, publish, and deploy content.
+  - However, `vets-website` is only deployed to production once per day.
+  - The CMS team can run content-only deploys independently of the daily production deploy, but that is triggered manually.
+- The content-only deploy does not include automated accessibility checks.
+- In local development, engineers can encounter confusing build errors that are due to the mismatch in outdated locally cached content and updated Liquid templates.
+
+### High Level Design
+
+There will be a new content-only auto-deploy job in addition to the existing full auto-deploy.
+- It will build content with CMS export data.
+- It will be a recurring job that executes on a more frequent schedule than the daily prod deploy.
+
+The schedule will run between 8am and 8pm ET to align with CMS support hours.
```

Comment:
The initial schedule

Reply:
The hourly cadence is the initial configuration, but I don't think we've discussed changing the timeframe of 8am to 8pm to something else as we roll out.

Reply:
I think Dror is just saying that the hourly cadence could, and presumably will, change as part of future work. The initial schedule is hourly, 8-8 on weekdays. Future schedule will be more frequent than hourly and the timeframe may expand.

Reply:
I've captured that the hourly cadence is likely to change, though I wasn't under the impression that the timeframe could expand too. I've updated this line to reflect that possibility.